plugins/BEdita/Core/src/Model/Behavior/CustomPropertiesBehavior.php

```diff
@@ -337,47 +340,129 @@ public function findCustomProp(SelectQuery $query, mixed ...$args): SelectQuery
+            foreach ($value as $operator => $v) {
+                if (is_numeric($operator)) {
+                    $in[] = $this->expressionValue($v, $schema, $driver);
+                    continue;
+                }
+
+                if ($operator === 'in' || $operator === 'notin' || $operator === 'nin') {
+                    $v = is_array($v) ? $v : [$v];
+                    $expValue = array_map(fn($i) => $this->expressionValue($i, $schema, $driver), $v);
+                } else {
+                    $expValue = $this->expressionValue($v, $schema, $driver);
+                }
+                $conditions[$key] = [$operator => $expValue];
+            }
+
+            if (!empty($in)) {
+                $conditions[$key] = ['in' => $in];
+            }
         }
-        unset($value);
-
-        return $query->where(function (QueryExpression $exp, SelectQuery $query) use ($options, $driver) {
-            $field = $this->table()->aliasField($this->getConfig('field'));
-
-            return $exp->and(array_map(
-                function ($key, $value) use ($field, $query, $driver) {
-                    if ($driver instanceof Mysql) {
-                        // MySQL syntax using JSON_EXTRACT and JSON_UNQUOTE
-                        return $query->expr()->eq(
-                            new FunctionExpression(
-                                'JSON_UNQUOTE',
-                                [
-                                    new FunctionExpression(
-                                        'JSON_EXTRACT',
-                                        [$field => 'identifier', sprintf('$.%s', $key)],
-                                    ),
-                                ],
-                            ),
-                            new FunctionExpression('JSON_UNQUOTE', [json_encode($value)]), // trick to normalize values compared
-                        );
-                    }
-
-                    // PostgreSQL syntax with native JSON column using ->> operator
-                    // For PostgreSQL's ->> operator, we need to handle value formatting correctly
-                    $compareValue = is_string($value) ? $value : json_encode($value); // Use json_encode to handle formatting
-
-                    // Use ->> operator directly on JSON column
-                    return $query->expr()->eq(
-                        sprintf('%s->>%s', $field, $query->getConnection()->getDriver()->quote($key)),
-                        $compareValue,
-                    );
-                },
-                array_keys($options),
-                $options,
-            ));
-        });
+
+        return $conditions;
+    }
+
+    /**
+     * Get expression for a property field.
+     *
+     * @param string $field Field name.
+     * @param string $key Property name.
+     * @param object $driver Database driver.
+     * @return mixed
+     */
+    protected function expressionField(string $field, string $key, object $driver): mixed
```

Comment:
```suggestion
    protected function expressionField(string $field, string $key, object $driver): FunctionExpression|string
```
If I'm not wrong it returns `FunctionExpression` or `string`